frontend/src/utils/generate-warnings.ts

```diff
@@ -89,6 +571,69 @@ function addCoursesToTracker(
   }
 }
 
+/**
+ * Add courses from a ScheduleTerm to a Map of coursecode to creditHours
+ * @param toAdd the ScheduleTerm
+ * @param taken the Map
+ */
+function addCoursesToMap(
+  toAdd: ScheduleTerm,
+  taken: Map<string, number>
+): void {
+  for (const course of toAdd.classes) {
+    taken.set(produceCourseCode(course), course.numCreditsMin);
+  }
+}
+
+/**
+ * Check if a course is in one of the subject ranges
+ * @param courseCode the course to check
+ * @param ranges the ranges to check against.
+ */
+function courseInSubjectRanges(
+  courseCode: string,
+  ranges: ISubjectRange[]
+): boolean {
+  let splitArr: string[] = courseCode.split(" ");
+  if (splitArr.length < 2) {
+    return false;
+  }
```

Comment:
I don't like this. I would rather store an object with a separate classId and subject.